state-chain/pallets/cf-elections/src/electoral_systems/blockchain/delta_based_ingress.rs

```diff
@@ -261,29 +267,64 @@ where
 				}
 			}
 
-			let mut election_access = electoral_access.election_mut(election_identifier)?;
-			if !closed_channels.is_empty() {
-				for closed_channel in closed_channels {
-					pending_ingress_totals.remove(&closed_channel);
-					channels.remove(&closed_channel);
-				}
+			for closed_channel in closed_channels {
+				pending_ingress_totals.remove(&closed_channel);
+				channels.remove(&closed_channel);
+			}
 
-				if channels.is_empty() {
-					election_access.delete();
-				} else {
-					election_access.set_state(pending_ingress_totals)?;
-					election_access.refresh(
-						// This value is meaningless. We increment as it is required to use a new
-						// higher value to refresh the election.
-						election_identifier
-							.extra()
-							.checked_add(1)
-							.ok_or_else(CorruptStorageError::new)?,
-						channels,
+			if channels.is_empty() {
+				let election_access = electoral_access.election_mut(election_identifier)?;
+				election_access.delete();
+			} else {
+				let properties = {
+					let mut election_access = electoral_access.election_mut(election_identifier)?;
+					election_access.set_state(pending_ingress_totals.clone())?;
+					election_access.properties()?
+				};
+
+				// calculate new properties by:
+				//  - taking accounts and channel_details from `channels`, keeping the preexisting
+				//    logic ƒor closing channels
+				//  - taking the consensus amount from either:
+				//     - the unsynchronized state map
+				//     - the election state
+				//    where we use the one which is higher.
+				let new_properties = channels
+					.iter()
+					.map(|(account, (channel_details, _))| {
+						let ingressed_amount = electoral_access
+							.unsynchronised_state_map(&(account.clone(), channel_details.asset))
+							.unwrap()
+							.unwrap_or_else(|| properties.get(account).unwrap().1);
+						let pending_ingressed_amount = pending_ingress_totals
+							.get(account)
+							.copied()
+							.unwrap_or_else(|| properties.get(account).unwrap().1);
```

Comment:
Isn't properties[account].1 just the ignored `_` from line 294? Ie. can we just rename the `_` to `old_properties` and use `unwrap_or(old_properties)` here?